workflows/cloudify_system_workflows/snapshots/credentials.py

```diff
@@ -182,8 +149,38 @@ def _create_agent_key_path_dict(self):
             deployment_id = elem[1]
             node_properties = pickle.loads(elem[2])
             agent_config = self._get_agent_config(node_properties)
+            tenant = elem[3]
             if 'key' in agent_config:
                 agent_key_path = agent_config['key']
                 key = deployment_id + '_' + node_id
-                agent_key_path_dict[key] = os.path.expanduser(agent_key_path)
+                agent_key_path_dict.setdefault(
+                    key, {})[tenant] = os.path.expanduser(agent_key_path)
         return agent_key_path_dict
+
+
+def add_key_secret(tenant, key_path, key_data):
+    key_name = SECRET_STORE_AGENT_KEY_PREFIX + ''.join(
+        char if char in ALLOWED_KEY_CHARS else '_'
+        for char in key_path
+        )
+
+    client = CloudifyClient(
+        'localhost',
```

Comment:
I meant that there's probably a way to get a REST client from some helper method which already does that..